src/main/kotlin/assignment/Assignment.kt

```diff
@@ -14,20 +18,63 @@ data class Response(
     val covidPositive: Boolean
 )
 
+fun String.toBool(): Boolean {
+    return this.uppercase()=="YES";
+}
+
 fun main() {
     val responses = readResponses()
     val processedData = processResponses(responses)
     writeToFile(processedData)
 }
 
 fun readResponses(): List<Response> {
-    TODO()
+    val fileName = "src/main/kotlin/assignment/responses.txt"
+    val lines: List<String> = File(fileName).readLines()
+
+    var responses = mutableListOf<Response>()
+    for (i in 1 until lines.size) {
+        var line = lines[i]
+        line = line.replace("\\s+".toRegex(), " ")
+        var eachResponse = line.split(" ")
+        responses.add(Response(eachResponse[2], eachResponse[3].toInt(), eachResponse[4], eachResponse[5].toBool()))
+    }
+    return responses.toList()
 }
 
 fun processResponses(responses: List<Response>): List<List<Response>> {
-    TODO()
+    val covidNegative = responses.filter { !it.covidPositive }
+    val sortedCovidNegative = covidNegative.sortedWith(compareBy({it.branch},{it.name}))
+
+    val groupByBranch = sortedCovidNegative.groupBy { it.branch }
+
+    var allResponseBranchWise = mutableListOf<List<Response>>()
+    groupByBranch.forEach { (k, v) ->
+        allResponseBranchWise.add(v)
+    }
+    var threedList = allResponseBranchWise.chunked(3)
+    var req = mutableListOf<List<Response>>()
+    threedList.forEach { req.add(it.flatten()) }
```

Comment:
Here also, you could have used the `map` function and avoided this new `MutableList`.